pkg/block/adapter.go

```diff
@@ -4,15 +4,21 @@ import (
 	"context"
 	"io"
 	"net/http"
-
-	"github.com/aws/aws-sdk-go/service/s3"
 )
 
+// MultipartPart single multipart information
+type MultipartPart struct {
+	ETag       string
+	PartNumber int64
```

Comment:
Any point in `int64` here?  The [docs](https://docs.aws.amazon.com/AmazonS3/latest/userguide/mpuoverview.html#mpu-process) say it's a number between 1 and 10_000...

Reply:
no point - kept the AWS types - but it will be easier to use int, we do not use code-generators yet.